src/abstractmcmc.jl

```diff
@@ -30,6 +30,15 @@ getadaptor(state::HMCState) = state.adaptor
 getmetric(state::HMCState) = state.metric
 getintegrator(state::HMCState) = state.κ.τ.integrator
 
+function AbstractMCMC.getparams(state::HMCState)
+    # TODO(sunxd): should we return a copy?
+    return state.transition.z.θ
+end
+
+function AbstractMCMC.setparams!!(state::HMCState, θ)
```

Comment:
One dangerous aspect there is that `state.transition` _caches_ the logjoint and gradient computations. So when you use the `@set` macro here, you're going to also keep the cached log-joint and gradient computation, which will then be out of sync with the parameters.

If you then naively pass this transition somewhere, say, into the next `step` call, AHMC.jl will use the incorrect logjoint eval in the MH step.

IMO the safe way is to use the explicit constructor of `PhasePoint` I believe without passing in the cached values. This should result in receomputation of this.

Reply:
just realized that `model` is not passed into `setparams!!`, I think for now, we only set the parameters, then when use the transition, logp should be recomputed. (we can also later introduce some like `setlogp` or `compute_logp` etc.) I'll add a comment in the code